lib/spack/spack/patch.py

```diff
@@ -36,10 +36,11 @@ def apply_patch(stage, patch_path, level=1, working_dir="."):
     """
     git_utils_path = os.environ.get("PATH", "")
     if sys.platform == "win32":
-        git = which_string("git", required=True)
-        git_root = git.split("\\")[:-2]
-        git_root.extend(["usr", "bin"])
-        git_utils_path = os.sep.join(git_root)
+        git = which_string("git")
+        if git:
+            git_root = git.split(os.sep)[:-2]
+            git_root.extend(["usr", "bin"])
+            git_utils_path = os.pathsep.join([os.sep.join(git_root), git_utils_path])
```

Comment:
Is this because git does not in fact always come with `patch`, as previously thought?

Reply:
Previously (and currently) in the getting started guide we explicitly call out "Git for Windows", which does vendor patch, and covers the _vast_ majority of Windows users to the point where most package managers don't bother to even have it as a package for Windows (conan and vcpkg).
However it is possible to install Git via other tools (although I had never heard of anyone doing so until that issue) so we should support that. 

Additionally, I see no reason _not_ to do this, why prevent Spack from being more flexible in its ability to load the tools it depends on.